apps/synapse-portal-framework/src/components/HeaderSearchBox.tsx

```diff
@@ -33,12 +37,20 @@ const HeaderSearchBox = ({
   roles,
 }: HeaderSearchBoxProps) => {
   const [role, setRole] = useState('')
-  const [, setSearchParams] = useSearchParams()
+  const [searchParams, setSearchParams] = useSearchParams()
   const theme = useTheme()
 
   const handleTermClick = (term: string) => {
     const trimmedTerm = term.trim()
-    setSearchParams({ FTS_SEARCH_TERM: trimmedTerm, role: role || '' })
+    const newSearchParams = new URLSearchParams(searchParams)
+    newSearchParams.set(FTS_SEARCH_TERM, trimmedTerm)
+
+    if (role) {
+      newSearchParams.set(FTS_SEARCH_ROLE, role)
+    }
+
+    setSearchParams(newSearchParams)
```

Comment:
Pass a callback to the state setter to ensure consistency

```suggestion
    setSearchParams(prev => {
  	    prev.set(FTS_SEARCH_TERM, trimmedTerm)
  	    if (role) {
	      prev.set(FTS_SEARCH_ROLE, role)
    	}
    	return prev
  	})
```